alarm/PolicyManager2.js

```diff
@@ -2118,46 +2118,57 @@ class PolicyManager2 {
         case 'create':
           for (const rawPolicy of rawData) {
             const { policy, alreadyExists } = await this.checkAndSaveAsync(new Policy(rawPolicy));
-            let result = policy;
-            if (alreadyExists == 'duplicated') {
-              result = 'duplicated'
-            }
-            results[action].push(result);
+            results[action].push({
+              policy: policy,
+              error: alreadyExists == 'duplicated'
+            });
           }
           break;
         case 'update':
           for (const rawPolicy of rawData) {
             const pid = rawPolicy.pid;
-            const oldPolicy = await this.getPolicy(pid);
-            await this.updatePolicyAsync(rawPolicy);
-            const newPolicy = await this.getPolicy(pid);
-            this.tryPolicyEnforcement(newPolicy, 'reenforce', oldPolicy);
-            results[action].push(newPolicy);
+            let newPolicy;
+            try {
+              const oldPolicy = await this.getPolicy(pid);
+              await this.updatePolicyAsync(rawPolicy);
+              newPolicy = await this.getPolicy(pid);
+              this.tryPolicyEnforcement(newPolicy, 'reenforce', oldPolicy);
+            } catch (e) {
+              log.warn(`update policy ${rawPolicy} error`,e);
+            }
+            results[action].push({
+              policy: newPolicy,
```

Comment:
newPolicy will be `undefined` in case of error

Reply:
also, do we need more error status than true/false?

Reply:
Yes, it will be undefined in case of error, so the error will be true. Keep `policy:batch` response data structure same would be helpful.